tapir/shifts/services/slot_modification_service.py

```diff
@@ -0,0 +1,226 @@
+import csv
+import datetime
+import io
+from dataclasses import dataclass
+from typing import List
+
+from django.db import transaction
+from django.db.models import Q
+from django.utils import timezone
+
+from tapir.accounts.models import TapirUser
+from tapir.shifts.models import (
+    ShiftTemplate,
+    ShiftSlotTemplate,
+    ShiftAttendanceTemplate,
+    ShiftAttendance,
+    ShiftSlot,
+)
+from tapir.utils.user_utils import UserUtils
+
+
+class SlotModificationService:
+    WORKDAY = "workday"
+    WEEKEND = "weekend"
+
+    @dataclass(frozen=True)
+    class ParameterSet:
+        workday_or_weekend: str
+        time: datetime.time
+        origin_slot_name: str
+        target_slot_name: str | None  # if None, the slot will be deleted
+        target_capabilities: frozenset | None
+
+    class SlotNames:
+        WARENANNAHME = "Warenannahme & Lager"
+        KASSE = "Kasse"
+        REINIGUNG = "Reinigung & Aufräumen"
+        ALLGEMEIN = ""
+
+    @classmethod
+    def build_changes(
+        cls,
+        parameter_sets: List[ParameterSet],
+        excluded_shift_template_ids: list[int] | None = None,
+    ):
+        return {
+            parameter_set: cls.pick_slots_to_modify(
+                parameter_set, excluded_shift_template_ids
+            )
+            for parameter_set in parameter_sets
+        }
+
+    @classmethod
+    def preview_changes(
```

Comment:
Seems this method is not used, did you add it for future use?

Reply:
Yes I thought we may want to see if a change would affect anyone before applying the changes.